modules/proxistoreBidAdapter.js

```diff
@@ -1,62 +1,86 @@
-  const sizeIds = [];
-
+  var sizeIds = [];
   bidRequests.forEach(function (bid) {
-    const sizeId = {
+    var sizeId = {
       id: bid.bidId,
       sizes: bid.sizes.map(function (size) {
         return {
           width: size[0],
-          height: size[1]
+          height: size[1],
         };
-      })
+      }),
+      floor: _assignFloor(bid),
+      segments: _assignSegments(bid),
     };
     sizeIds.push(sizeId);
   });
-  const payload = {
+  var payload = {
     auctionId: bidRequests[0].auctionId,
     transactionId: bidRequests[0].auctionId,
     bids: sizeIds,
     website: bidRequests[0].params.website,
     language: bidRequests[0].params.language,
     gdpr: {
-      applies: false
-    }
-  };
-  const options = {
-    contentType: 'application/json',
-    withCredentials: true
+      applies: false,
+    },
   };
 
   if (bidderRequest && bidderRequest.gdprConsent) {
-    if (typeof bidderRequest.gdprConsent.gdprApplies === 'boolean' && bidderRequest.gdprConsent.gdprApplies) {
+    if (
+      typeof bidderRequest.gdprConsent.gdprApplies === 'boolean' &&
+      bidderRequest.gdprConsent.gdprApplies
+    ) {
       payload.gdpr.applies = true;
     }
 
-    if (typeof bidderRequest.gdprConsent.consentString === 'string' && bidderRequest.gdprConsent.consentString) {
+    if (
+      typeof bidderRequest.gdprConsent.consentString === 'string' &&
+      bidderRequest.gdprConsent.consentString
+    ) {
       payload.gdpr.consentString = bidderRequest.gdprConsent.consentString;
     }
 
-    if (bidderRequest.gdprConsent.vendorData && bidderRequest.gdprConsent.vendorData.vendorConsents && typeof bidderRequest.gdprConsent.vendorData.vendorConsents[PROXISTORE_VENDOR_ID.toString(10)] !== 'undefined') {
-      payload.gdpr.consentGiven = !!bidderRequest.gdprConsent.vendorData.vendorConsents[PROXISTORE_VENDOR_ID.toString(10)];
+    if (
+      bidderRequest.gdprConsent.vendorData &&
+      bidderRequest.gdprConsent.vendorData.vendorConsents &&
+      typeof bidderRequest.gdprConsent.vendorData.vendorConsents[PROXISTORE_VENDOR_ID.toString(10)] !== 'undefined'
+    ) {
+      payload.gdpr.consentGiven = !!bidderRequest.gdprConsent.vendorData
+        .vendorConsents[PROXISTORE_VENDOR_ID.toString(10)];
     }
   }
 
+  const options = {
+    contentType: 'application/json',
+    withCredentials: !!payload.gdpr.consentGiven,
+  };
+  const endPointUri = payload.gdpr.consentGiven
+    ? `https://abs.proxistore.com/${payload.language}/v3/rtb/prebid/multi`
+    : `https://cookieless-proxistore.com/v3/rtb/prebid/multi/cookieless`;
+
   return {
     method: 'POST',
-    url: bidRequests[0].params.url || 'https://abs.proxistore.com/' + payload.language + '/v3/rtb/prebid/multi',
+    url: endPointUri,
     data: JSON.stringify(payload),
-    options: options
+    options: options,
   };
 }
 
+function _assignSegments(bid) {
+  if (bid.hasOwnProperty('fpd')) {
+    return bid.fpd.segments || [];
```

Comment:
First party data is not provided through `fpd` anymore, there's ortb2 interface now, you can read more about it here: https://docs.prebid.org/features/firstPartyData.html